### ISSUE TITLE
jQuery.sap.extend() should be replaced or not?

### ISSUE THREAD

**Issue by NehKulkarni** [SAP/ui5-migration#59]
Hi, we use SAPUI5 version 1.60.3 for our enterprise application. While moving to AMD-like syntax, we have currently replaced shallow and deep jQuery.sap.extend() with merge(), as Object.assign() resulted in errors as it did not copy prototype properties. The recent push to ui5-migration tool suggested we do not replace it at all. Whereas the page https://openui5.hana.ondemand.com/1.60.3//#/topic/a075ed88ef324261bca41813a6ac4a1c still suggests the otherwise. Could anyone just throw some light as to what needs to be done? Thanks!

**Comment by tommyvinhlam** [comment]
Hi @NehKulkarni,

with the latest push #58 we have **disabled** the automatic replacement of `jQuery.sap.extend` in the migration tool. This means, that `jQuery.sap.extend` needs to be migrated manually and does not mean, that jQuery.sap.extend shouldn't be replaced at all.
The replacement with Object.assign() was not always suitable w.r.t. not copying prototype properties and might break applications which rely on this behaviour (See #55).

But let me tell you, that we are currently working on this topic. With https://github.com/SAP/openui5/commit/1203b81613ef82e354a4ce70c4268885792401c3 we have introduced new modules for shallow and deep copy, which can be used instead.
Also we have updated the documentation regarding the legacy jQuery.sap replacements with our latest recommendations. These updates will be available with the next release (version 1.71).

I hope this helps you out.
